sotodlib/preprocess/processes.py

```diff
@@ -585,6 +620,49 @@ class SubPolyf(_Preprocess):
     def process(self, aman, proc_aman):
         tod_ops.sub_polyf.subscan_polyfilter(aman, **self.process_cfgs)
 
+class SSOFootprint(_Preprocess):
+    """Find nearby sources within a given distance and get SSO footprint and plot
+    each source on the focal plane.
+
+    .. autofunction:: sotodlib.obs_ops.sources.get_sso
+    """
+    name = 'sso_footprint'
+
+    def calc_and_save(self, aman, proc_aman):
+        ssos = planets.get_nearby_sources(tod=aman, distance=self.calc_cfgs.get("distance", 20))
+        if ssos:
+            sso_aman = core.AxisManager()
+            for sso in ssos:
+                planet = sso[0]
+                xi_p, eta_p = obs_ops.sources.get_sso(aman, planet, nstep=self.calc_cfgs.get("nstep", 100))
+                planet_aman = core.AxisManager()
+                planet_aman.wrap("xi_p", xi_p)
+                planet_aman.wrap("eta_p", eta_p)
+                sso_aman.wrap(planet, planet_aman)
+            self.save(proc_aman, sso_aman)
+        else:
+            raise ValueError("No sources found within footprint")
+        
+    def save(self, proc_aman, sso_aman):
+        if self.save_cfgs is None:
+            return
+        if self.save_cfgs:
+            proc_aman.wrap("sso_footprint", sso_aman)
+
+    def plot(self, aman, proc_aman, filename):
+        if self.plot_cfgs is None:
+            return
+        if self.plot_cfgs:
+            from .preprocess_plot import plot_sso_footprint
+            filename = filename.replace('{ctime}', f'{str(aman.timestamps[0])[:5]}')
+            filename = filename.replace('{obsid}', aman.obs_info.obs_id)
+            for sso in proc_aman.sso_footprint.keys():
```

Comment:
What you want here is `proc_aman.sso_footprint._assignment.keys()` then you shouldn't need the `sso == 'dets' or ssh == 'samps'` check.